src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DkmUtilities.cs

```diff
@@ -68,13 +74,28 @@ internal unsafe static ImmutableArray<MetadataBlock> GetMetadataBlocks(this DkmC
                 }
                 Debug.Assert(block.ModuleVersionId == module.Mvid);
                 builder.Add(block);
+                index++;
             }
             // Include "intrinsic method" assembly.
             ptr = runtime.GetIntrinsicAssemblyMetaDataBytesPtr(out size);
-            builder.Add(GetMetadataBlock(ptr, size));
+            builder.Add(GetMetadataBlock(previousMetadataBlocks, index, ptr, size));
             return builder.ToImmutableAndFree();
         }
 
+        private static MetadataBlock GetMetadataBlock(ImmutableArray<MetadataBlock> previousMetadataBlocks, int index, IntPtr ptr, uint size)
+        {
+            if (previousMetadataBlocks != null && !previousMetadataBlocks.IsDefault && previousMetadataBlocks.Length > index)
```

Comment:
nit: presonally i think it would read better if it was "index < previousMetadataBlocks.Length"